common/reflect_cache.go

```diff
@@ -0,0 +1,31 @@
+package common
+
+import (
+	"reflect"
+	"sync"
+)
+
+var mutex sync.Mutex
+
+var mapCache = map[reflect.Type]map[string]string{}
+
+func getJsonToFieldNameMap(structType reflect.Type) map[string]string {
+	if structType.Kind() != reflect.Ptr && structType.Kind() != reflect.Struct {
+		return nil
+	}
+	mutex.Lock()
+	defer mutex.Unlock()
```

Comment:
This is probably fine, but if we want to slightly decrease lock contention here, we can actually release the lock while we're doing reflection and acquire it again only when adding the value to the cache.